### ISSUE TITLE
flatten and unflatten not the inverse

### ISSUE THREAD

**Issue by eptx** [amirziai/flatten#29]
`from flatten_json import flatten
from flatten_json import unflatten
data = {
    "a": 1,
    "b": 2,
    "c": [{"d": [2, 3, 4], "e": [{"f": 1, "g": 2}]}]
}

print(data)
flatData = flatten(data)
print(flatData)
unflat = unflatten(flatData)
print(unflat)`

output...

`{'a': 1, 'b': 2, 'c': [{'d': [2, 3, 4], 'e': [{'f': 1, 'g': 2}]}]}
{'a': 1, 'b': 2, 'c_0_d_0': 2, 'c_0_d_1': 3, 'c_0_d_2': 4, 'c_0_e_0_f': 1, 'c_0_e_0_g': 2}
{'a': 1, 'b': 2, 'c': {'0': {'d': {'0': 2, '1': 3, '2': 4}, 'e': {'0': {'f': 1, 'g': 2}}}}}`

**Comment by TheGreatKetchup** [comment]
Agreed, I have the same problem. Here is a slightly simpler example.
```python
>>> from pprint import print
>>> from flatten_json import flatten, unflatten
>>> x = {'a':['b', 'c']}
>>> pprint(flatten(x))
{'a_0': 'b', 'a_1': 'c'}
>>> pprint(unflatten(flatten(x)))
{'a': {'0': 'b', '1': 'c'}}
>>> 
```

The problems appears to be that the unflatten does not handle the reconstruction of lists at all.
